File: RecipeManagerWeb/Models/Recipe.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace RecipeManagerWeb.Models
{
    public class Recipe
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public int RecipeCategoryId { get; set; }
        [ForeignKey("RecipeCategoryId")]
        public RecipeCategory RecipeCategory { get; set; } = null!;
        public double Amount { get; set; } = 1.0;
        public PortionUnit? PortionUnit { get; set; }
        public int? Time { get; set; }
        public bool? Vegetarian { get; set; }
        public List<RecipeGroceryItem> Ingredients { get; set; } = new List<RecipeGroceryItem>();
        public List<InstructionStep> Instructions { get; set; } = new List<InstructionStep>();
        public string? Source { get; set; }
        public string? Comment { get; set; }

        public List<RecipeGroceryItem> GetIngredientsForRequestedAmount(double requestedAmount)
        {
            Func<double, double, double> division = (x, y) => x / y;
            Func<double, double, double> multiply = (x, y) => x * y;
            var standardizingOperator = (Amount > 1.0) ? division : multiply;
            Ingredients.ForEach(i => i.Amount = standardizingOperator(i.Amount, Amount));

            Ingredients.ForEach(i => i.Amount = i.Amount * requestedAmount);
            return Ingredients;
        }
    }

}

File: recipemanagerwebclient/lib/models/recipe.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:recipemanagerwebclient/models/base_model.dart';
import 'package:recipemanagerwebclient/models/data_layer.dart';

part 'recipe.g.dart';

@JsonSerializable()
class Recipe extends BaseModel {
  final int id;
  String name;
  late RecipeCategory recipeCategory;
  double amount;
  PortionUnit portionUnit;
  int time;
  bool vegetarian;
  late List<Ingredient> ingredients;
  late List<InstructionStep> instructions;
  String source;
  String comment;

  Recipe({
    this.id = 0,
    this.name = '',
    RecipeCategory? recipeCategory,
    this.amount = 0.0,
    this.portionUnit = PortionUnit.Bread,
    this.time = 0,
    this.vegetarian = false,
    List<Ingredient>? ingredients,
    List<InstructionStep>? instructions,
    this.source = '',
    this.comment = '',
  }) {
    this.recipeCategory = recipeCategory ?? RecipeCategory();
    this.ingredients = ingredients ?? <Ingredient>[];
    this.instructions = instructions ?? <InstructionStep>[];
  }

  factory Recipe.fromJson(Map<String, dynamic> json) => _$RecipeFromJson(json);
  static convertFromJson(Map<String, dynamic> json) => _$RecipeFromJson(json);
  @override
  Map<String, dynamic> toJson() => _$RecipeToJson(this);
}
